lib/figo.js

```diff
@@ -181,7 +192,58 @@ var HttpsAgent = function() {
   };
 
   return agent;
-}
+};
+
+
+// constant for number of attempts allowed to make to API, 1 initial and 2 retries, 3 total
+var ALLOWED_NUMBER_OF_ATTEMPTS = 3;
+
+// ### Initialization of https request with a retry count of three.
+//
+// Constructor parameters:
+//
+// - **agent** (`HttpsAgent`) - `HttpsAgent` object
+//
+// - **authorization** (`String`) - encoded authorization in https header
+//
+// - **contentType** (`String`) - content-type for https header
+//
+// - **path** (`String`) - the URL path on the server
+//
+// - **data** (`Object`) - If this parameter is defined, then it will be used as JSON-encoded POST content.
+//
+// - **method** (`Object`) - HTTP method (GET/POST/PUT/DELETE/etc.).
+//
+// - **stringify** (`Object`) - stringify function (either querystring.stringify or JSON.stringify).
+//
+// - **callback** (`Function`) - callback function with two parameters: `error` and `result`
+//
+var queryWithRetries = function (agent, authorization, contentType, contentLength, path, data, method, stringify, callback) {
```

Comment:
AFAIK: `agent` is the same through all the code base. So it can be just hardcoded and there will be no need to pass it as a parameter.
`stringify`, `contentLength` dont look like really necessary as a parameter as well.

Logic for authorization can be moved inside the function (like it was before).